hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -627,23 +633,26 @@ private double calculateAvgUtilization(List<DatanodeUsageInfo> nodes) {
     }
     clusterCapacity = aggregatedStats.getCapacity().get();
     clusterUsed = aggregatedStats.getScmUsed().get();
+    clusterRemaining = aggregatedStats.getRemaining().get();
 
-    return clusterUsed / (double) clusterCapacity;
+    return (clusterCapacity - clusterRemaining) / (double) clusterCapacity;
   }
 
   /**
-   * Calculates the utilization, that is used space divided by capacity, for
-   * the given datanodeUsageInfo.
+   * Calculates the utilization, that is (capacity - remaining) divided by
+   * capacity, for the given datanodeUsageInfo.
    *
    * @param datanodeUsageInfo DatanodeUsageInfo to calculate utilization for
    * @return Utilization value
    */
   public static double calculateUtilization(
       DatanodeUsageInfo datanodeUsageInfo) {
     SCMNodeStat stat = datanodeUsageInfo.getScmNodeStat();
+    double capacity = stat.getCapacity().get();
+    double remaining = stat.getRemaining().get();
+    double numerator = capacity - remaining;
 
-    return stat.getScmUsed().get().doubleValue() /
-        stat.getCapacity().get().doubleValue();
+    return numerator / capacity;
   }
```

Comment:
Is it possible to have function which is used by balancer as well as utilisation api?

Reply:
Currently the utilization api uses `compareByRemainingRatio` in `SCMNodeStat`. We could instead have a `calculateUtilization` method in the `DatanodeUsageInfo` class.